src/passes/references/dataAccessFunctionaliser.ts

```diff
@@ -309,7 +309,14 @@ export class DataAccessFunctionaliser extends ReferenceSubPass {
     }
 
     if (replacement !== null) {
-      this.replace(node, replacement, undefined, actualLoc, expectedLoc, ast);
+      this.replace(
+        node,
+        replacement,
+        undefined,
+        actualLoc,
+        expectedLoc ?? DataLocation.Default,
```

Comment:
The issue here is specific to ifStatements, not indexAccesses. An unused indexAccess should still have an undefined expected location, the problem was that expectedLocationAnalyser wasn't setting the expected location of the if condition to default. Could you modify expectedLocationAnalyser to do so rather than doing this

Reply:
Change is done.